src/main/controllers/OtpLoginController.ts

```diff
@@ -1,17 +1,59 @@
 import { Request, Response } from 'express';
+import {OtpActions} from "../resources/actions/otpActions";
+
+const email = 'pippo@pluto.com';
 
 export default class OtpLoginController {
   public get(req: Request, res: Response): void {
-    res.render('otp-login');
+    const otpAction = new OtpActions();
+    const attempts = otpAction.getAttempts(email);
+    if (attempts > 1)
+    {
+      res.render('otp-login', { invalidInputError: false, noValidCodeError: false, attemptCounter: attempts });
+    }
+    else
+    {
+      res.redirect('account-locked');
+    }
   }
 
   public post(req: Request, res: Response): void {
-    if (req.body['otp-code'] && req.body['otp-code'].length === 6) {
-      res.redirect('subscription-management');
-    } else {
-      res.redirect('otp-login');
-    }
+    const searchInput = req.body['otp-code'];
+    const otpAction = new OtpActions();
+    let attempts = otpAction.getAttempts(email);
+    if (attempts > 0)
+    {
+      if (searchInput && searchInput.length === 6 && searchInput.match(/^[0-9]+$/)) {
+        const searchInputNumber = parseInt(searchInput);
+        const otps = otpAction.validateOtp(searchInputNumber, email);
+        if (otps && otps.otpValid) {
+          otpAction.resetAttempts(email);
+          res.redirect('subscription-management');
+        }
+        else if(attempts > 1) {
+          attempts = otpAction.decrementAttempts(email);
+          res.render('otp-login', { invalidInputError: false, noValidCodeError: true, attemptCounter: attempts });
+        }
+        else {
+          res.redirect('account-locked');
+        }
 
+      } else {
+        attempts = otpAction.decrementAttempts(email);
```

Comment:
I'm wondering we we should be decrementing attempts in this scenario, as the code is invalid rather than incorrect

Reply:
noted and I agree. but it is not part of this task at the moment as we still don't know how it works on the azure side

Reply:
this is true but we know how basic requests work so we just need to match it to something closer to what it will be like